File: magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleAware.java
package com.wealthfront.magellan.lifecycle;

import android.content.Context;

import org.jetbrains.annotations.NotNull;

public interface LifecycleAware {
  
  default void create(@NotNull Context context) {}

  default void start(@NotNull Context context) {}

  default void resume(@NotNull Context context) {}

  default void pause(@NotNull Context context) {}

  default void stop(@NotNull Context context) {}

  default void destroy(@NotNull Context context) {}

  default boolean backPressed() {
    return false;
  }
}

File: magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleAware.kt
package com.wealthfront.magellan.lifecycle

import android.content.Context

interface LifecycleAware {

  @JvmDefault fun create(context: Context) {}

  @JvmDefault fun show(context: Context) {}

  @JvmDefault fun resume(context: Context) {}

  @JvmDefault fun pause(context: Context) {}

  @JvmDefault fun hide(context: Context) {}

  @JvmDefault fun destroy(context: Context) {}

  @JvmDefault fun backPressed() = false
}
